File: java-processor/src/GuardrailService.java
public final class GuardrailService {
    private static final String[] BLOCKED_TERMS = {
        "hack",
        "fraud",
        "scam",
        "steal",
        "money laundering"
    };

    public boolean isBlocked(String message) {
        String normalizedMessage = message.toLowerCase();

        for (String term : BLOCKED_TERMS) {
            if (normalizedMessage.contains(term)) {
                return true;
            }
        }

        return false;
    }
}

File: python-api/app/services/guardrail_service.py
class GuardrailService:
    BLOCKED_TERMS = (
        "hack",
        "fraud",
        "scam",
        "steal",
        "money laundering",
    )

    def is_blocked(self, message: str) -> bool:
        normalized_message = message.lower()
        return any(term in normalized_message for term in self.BLOCKED_TERMS)
